### PR TITLE
Add new pstext option to cast a shade beneath a text box

### PR DESCRIPTION
**Pull request by PaulWessel**
**Description of proposed changes**

This PR is made in support of #4703 by adding new option **-S**[_dx_/_dy_/][_shade_] to **pstext** so that either single or paragraph text may enjoy the optional shade-effect.  In the process, I did

1. Report an error if **-C+tc**|**C** is used with regular (i.e., not **-M**) text
2. Add a test to demonstrate that the option works for both types of text
3. Fix a minor issue in another test (book.sh)
4. Implement **-S** so that the backwards compatibility for GMT4 **-S**_pen_ still is honored.

As an example, this suggestion is now listed in the documentation:

`gmt text -R0/10/0/5 -Jx1c -F+f32p+cCM+tWELCOME -B -Gyellow -Wfaint -S -C+tO -pdf map`

with this result:

map

### CODE REVIEW

File: src/pstext.c [hunk 1]
```diff
@@ -356,6 +368,8 @@ static int usage (struct GMTAPI_CTRL *API, int level) {
 	GMT_Message (API, GMT_TIME_NONE, "\t-N Do Not clip text that exceeds the map boundaries [Default will clip].\n");
 	GMT_Option (API, "O,P");
 	GMT_Message (API, GMT_TIME_NONE, "\t-Q For all text to be (l)lower or (u)pper-case [Default leaves text as is].\n");
+	GMT_Message (API, GMT_TIME_NONE, "\t-S Plot a shadow behind the text box.  Requires -G<color> to be given as well.\n");
```
Comment (joa-quim) [inline src/pstext.c:371]: If it requires **-G** as well what is the **-S**_shade_ parameter used to?

Comment (PaulWessel) [reply src/pstext.c:371]: Text box may be white, yellow, or whatever, but the shade it casts need to be different for most cases.  See plot above.

**Review by joa-quim — APPROVED** [review]

**Comment by joa-quim** [comment]
Yes, realized that only after writing my comment.

**Comment by PaulWessel** [comment]
Need this to do so from movie but this is actually a nice option in text as is.

**Pull request merged**